Source/BugsnagMetadata.m

```diff
@@ -45,13 +45,30 @@ - (id)initWithDictionary:(NSDictionary *)dict {
     if (self = [super init]) {
         // Ensure that the instantiating dictionary is mutable.
         // Saves checks later.
-        self.dictionary = [dict mutableCopy];
+        self.dictionary = [self sanitizeDictionary:[dict mutableCopy]];
         self.stateEventBlocks = [NSMutableArray new];
     }
     [self notifyObservers];
     return self;
 }
 
+- (NSMutableDictionary *)sanitizeDictionary:(NSMutableDictionary *)input {
+    for (NSString *key in [input allKeys]) {
+        id obj = input[key];
+
+        if (obj == [NSNull null]) {
+            [input removeObjectForKey:key];
+        } else if ([obj isKindOfClass:[NSDictionary class]]) {
+            input[key] = [self sanitizeDictionary:[obj mutableCopy]];
+        } else if ([obj isKindOfClass:[NSArray class]]) {
+            NSMutableArray *ary = [obj mutableCopy];
```

Comment:
Presumably the array could contain dictionaries?

Reply:
Handled this case by recursing.

Reply:
I think you're still missing the case for an array of arrays:
```json
{
  [
    [
      { 'key': [NSNull null] },
      [NSNull null]
    ]
  ]
}
```

Reply:
That's not quite it - added suggested test below.